cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileService.java

```diff
@@ -214,7 +214,12 @@ public void deleteProfile(ProfileId profileId)
       AppMetadataStore appMetadataStore = AppMetadataStore.create(cConf, context, datasetFramework);
       deleteProfile(profileDataset, appMetadataStore, profileId, profile);
     }, NotFoundException.class, ProfileConflictException.class);
-    deleteMetrics(profileId);
+
+    try {
+      deleteMetrics(profileId);
+    } catch (IOException e) {
```

Comment:
seems a little cleaner to move the try/catch to deleteMetrics()?

Reply:
Good point